django/core/management/commands/shell.py

```diff
@@ -39,36 +44,47 @@ def add_arguments(self, parser):
 
     def _ipython_pre_011(self):
         """Start IPython pre-0.11"""
-        from IPython.Shell import IPShell
+        try:
+            from IPython.Shell import IPShell
+        except ImportError:
+            raise ShellImportError
```

Comment:
Adam, I guess you meant "classes rather than instances" since this comment isn't addressed?